File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/discovery/docs/HelloController.kt
package io.micronaut.discovery.docs

import io.micronaut.context.annotation.Requires
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

/**
 * The `hello-world` service the examples discover.
 */
@Requires(property = "spec.name", value = "HelloClientTest")
@Controller("/hello")
class HelloController {

    @Get
    fun index(): String = "Hello World"
}

File: test-suite-docs-python/src/test/python/micronaut/discovery/docs/HelloController.py
from micronaut.context.annotation import Requires
from micronaut.http.annotation import Controller, Get


# The hello-world service the examples discover.
@Requires(property="spec.name", value="HelloClientTest")
@Controller("/hello")
class HelloController:

    @Get
    def index(self) -> str:
        return "Hello World"
